packages/@aws-cdk/aws-ivs-alpha/lib/channel.ts

```diff
@@ -162,11 +200,16 @@ export class Channel extends ChannelBase {
       throw new Error(`channelName must contain only numbers, letters, hyphens and underscores, got: '${this.physicalName}'`);
     }
 
+    if (props.type && [ChannelType.STANDARD, ChannelType.BASIC].includes(props.type) && props.preset) {
+      throw new Error('preset cannot be used when STANDARD or BASIC channel type');
```

Comment:
Looking at the documentation [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ivs-channel.html) can it be an empty string for other two cases 
`For other channel types (BASIC and STANDARD), preset is the empty string ("")`

Reply:
@shikha372 
Thank you for the review.
I was aware of that. However, since preset cannot be used for BASIC or STANDARD, an empty string for preset is essentially the same as undefined.
Therefore, in the L2 construct, I allowed setting the preset only for the Advanced type.

If you have a better suggestion for improvement, could you please let me know?
Currently, I'm considering the following two solutions:

* Add an equivalent of an empty string to the enum, and allow setting it only for BASIC or STANDARD.
* If the preset is set for BASIC or STANDARD, overwrite it with an empty string.

However, neither of these seems like a good solution, and I'm struggling with it.​​​​​​​​​​​​​​​​..

Reply:
Thanks @mazyu36 , i was thinking of second one that is if preset is 'BASIC' or 'STANDARD', can we overwrite with an empty string?

Reply:
@shikha372 
Thanks for the comment. When specifying the preset with the STANDARD or BASIC channel type, I have modified it to overwrite the preset with an empty string. I have also updated the docs and added unit test and integration test cases.

Reply:
Thank you @mazyu36